opentracing-vertx-web/src/main/java/io/opentracing/contrib/vertx/ext/web/TracingHandler.java

```diff
@@ -0,0 +1,118 @@
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        if (routingContext.failed()) {
+            handlerFailure(routingContext);
+        } else {
+            handlerNormal(routingContext);
+        }
+    }
+
+    protected void handlerNormal(RoutingContext routingContext) {
+        // reroute
+        Object object = routingContext.get(ACTIVE_SEVER_SPAN);
+        if (object != null) {
+            Span span = (Span) object;
+            decorators.forEach(spanDecorator ->
+                    spanDecorator.onReroute(routingContext.request(), span));
+
+            // TODO in 3.3.3 it was sufficient to add this when creating the span
+            routingContext.addBodyEndHandler(finishEndHandler(routingContext, span));
+            routingContext.next();
+            return;
+        }
+
+        SpanContext extractedContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
+                new MultiMapExtractAdapter(routingContext.request().headers()));
+
+        Span span = tracer.buildSpan(routingContext.request().method().toString())
+                .asChildOf(extractedContext)
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+                .startManual();
+
+        decorators.forEach(spanDecorator ->
+                spanDecorator.onRequest(routingContext.request(), span));
+
+        routingContext.put(ACTIVE_SEVER_SPAN, span);
+        routingContext.addBodyEndHandler(finishEndHandler(routingContext, span));
+        routingContext.next();
+    }
+
+    protected void handlerFailure(RoutingContext routingContext) {
+        Object object = routingContext.get(ACTIVE_SEVER_SPAN);
+        final Span span = object instanceof Span ? (Span) object : null;
+        if (span != null) {
+            routingContext.addBodyEndHandler(event -> decorators.forEach(spanDecorator ->
+                    spanDecorator.onFailure(routingContext.failure(), routingContext.response(), span)));
+        }
+
+        routingContext.next();
+    }
+
+    private Handler<Void> finishEndHandler(RoutingContext routingContext, Span span) {
+        return handler -> {
+            decorators.forEach(spanDecorator ->
+                    spanDecorator.onResponse(routingContext.request(), span));
+            span.finish();
+        };
+    }
+
+    /**
+     * Helper method for accessing server span context associated with current request.
+     *
+     * @param routingContext routing context
+     * @return server span context or null if not present
+     */
+    public static SpanContext serverSpanContext(RoutingContext routingContext) {
+        SpanContext serverContext = null;
+
+        Object object = routingContext.get(ACTIVE_SEVER_SPAN);
+        if (object instanceof Span) {
+            Span span = (Span) object;
+            serverContext = span.context();
+        } else {
+            log.error("severSpanContext is null or not an instance of SpanContext");
```

Comment:
`serverSpanContext` - although may be better to change to `Server SpanContext is,,,,`